lib/system/gc_common.nim

```diff
@@ -144,15 +144,35 @@ when allowForeignThreadGc:
     ## This function is available only when ``--threads:on`` and ``--tlsEmulation:off``
     ## switches are used
     if not localGcInitialized:
+      if not gch.stackBottom.isNil:
```

Comment:
I have a dll written in Nim with some API. Each API function calls setupForeignThreadGc. But I have no guarantees that API will not be used in the main thread, that initialized via ``NimMain``, or in another Nim thread (via callback). We must allow to call ``setupForeignThreadGc`` and ``tearDownForeignThreadGc`` in the Nim thread.